lms/djangoapps/bulk_enroll/views.py

```diff
@@ -72,9 +95,32 @@ def post(self, request):
                 'action': serializer.data.get('action'),
                 'courses': {}
             }
-            for course in serializer.data.get('courses'):
-                response = students_update_enrollment(self.request, course_id=course)
-                response_dict['courses'][course] = json.loads(response.content)
+            for course_id, cohort_name in itertools.izip_longest(serializer.data.get('courses'),
+                                                                 serializer.data.get('cohorts', [])):
+                response = students_update_enrollment(self.request, course_id=course_id)
+                response_content = json.loads(response.content)
+
+                if cohort_name:
+                    course_key = CourseKey.from_string(course_id)
+                    cohort = get_cohort_by_name(course_key=course_key, name=cohort_name)
+                    for user_data in response_content['results']:
+                        if "after" in user_data and user_data["after"].get("enrollment", False) is True or \
```

Comment:
This is missing brackets. Should be:
`if "after" in user_data and (user_data["after"].get("enrollment", False) is True or user_data["after"].get("allowed", False) is True):`